src/plugins/mer/merdeployconfiguration.cpp

```diff
@@ -158,8 +158,11 @@ void MerAddRemoveSpecialDeployStepsProjectListener::timerEvent(QTimerEvent *even
 {
     if (event->timerId() == m_updateTargetsTimer.timerId()) {
         m_updateTargetsTimer.stop();
-        while (!m_updateTargetsQueue.isEmpty())
-            updateTarget(m_updateTargetsQueue.dequeue());
+        while (!m_updateTargetsQueue.isEmpty()) {
+            const TargetPtr target = m_updateTargetsQueue.dequeue();
+            if (!target.isNull())
+                updateTarget(target);
+        }
```

Comment:
I would prefer not to introduce TargetPtr at the scope of this class, leave the `while` loop as is and simply prepend:

```
m_updateTargetsQueue.removeAll(nullptr); // destroyed meanwhile
```